miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,73 @@
+use crate::assembler::assembler;
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::{dsa::rpo_falcon512, merkle::MerkleStore},
+    utils::vec,
+    AccountError, Word, ZERO,
+};
+
+pub enum AuthScheme {
+    RpoFalcon512 { pub_key: rpo_falcon512::PublicKey },
+}
```

Comment:
We have the same enum for basic wallet. Instead of duplicating it here, I'd move it to some central place (maybe into `src/auth.rs`?).

Also, since this is a part of the public interface, we should add some doc comments here.